barretenberg/cpp/src/barretenberg/crypto/merkle_tree/node_store/cached_content_addressed_tree_store.hpp

```diff
@@ -926,32 +970,39 @@ void ContentAddressedCachedTreeStore<LeafValueType>::unwind_block(const block_nu
                                             forkConstantData_.name_));
         }
     }
-    WriteTransactionPtr writeTx = create_write_transaction();
-    try {
-        // std::cout << "Removing block " << blockNumber << std::endl;
-
-        // Remove the block's node and leaf data given the max index of the previous block
-        std::optional<index_t> maxIndex = std::optional<index_t>(previousBlockData.size);
-        remove_node(std::optional<fr>(blockData.root), 0, maxIndex, *writeTx);
-        // remove the block from the block data table
-        dataStore_->delete_block_data(blockNumber, *writeTx);
-        dataStore_->delete_block_index(blockData.size, blockData.blockNumber, *writeTx);
-        uncommittedMeta.unfinalisedBlockHeight = previousBlockData.blockNumber;
-        uncommittedMeta.size = previousBlockData.size;
-        uncommittedMeta.committedSize = previousBlockData.size;
-        uncommittedMeta.root = previousBlockData.root;
-        // std::cout << "New block root " << previousBlockData.root << std::endl;
-        //  commit this new meta data
-        persist_meta(uncommittedMeta, *writeTx);
-        writeTx->commit();
-    } catch (std::exception& e) {
-        writeTx->try_abort();
-        throw std::runtime_error(format("Unable to commit unwind of block: ",
-                                        blockNumber,
-                                        ". Tree name: ",
-                                        forkConstantData_.name_,
-                                        " Error: ",
-                                        e.what()));
+
+    {
+        WriteTransactionPtr writeTx = create_write_transaction();
+        try {
+            // std::cout << "Removing block " << blockNumber << std::endl;
+
+            // If the tree was empty at the block being removed then we should not attempt to remove
+            // any nodes. (there were no nodes at the point this block was comitted)
+            if (blockData.size > 0) {
```

Comment:
Here is the other part of the fix. We don't attempt to remove nodes if at this point the tree was still empty.